backend/models/group.js

```diff
@@ -785,46 +776,168 @@ function findObjectsByQuery(account, model, query) {
 	});
 }
 
-function findModelSharedIDsByQuery(account, model, query, branch, revId) {
-	return findObjectsByQuery(account, model, query).then((results) => {
-		if (results && results.length > 0) {
-			return History.getHistory({ account, model }, branch, revId).then((history) => {
-				if (!history) {
-					return Promise.reject(responseCodes.INVALID_TAG_NAME);
-				} else {
-					return db.getCollection(account, model + ".scene").then((dbCol) => {
-						const parents = results.map(x => x = x.parents).reduce((acc, val) => acc.concat(val), []);
-
-						// NOTE: we've seen parents.length >15000 has caused a failure on the database. so I added
-						// a loop to restrict the length of parents per query. This needs to be revisited when
-						// we've done the optimisation for revId
-						const entryPerQuery = 7000;
-						const queryPromise = [];
-						for(let i = 0; i < parents.length; i = i + entryPerQuery) {
-							const endIndex = i + entryPerQuery < parents.length ? i + entryPerQuery : parents.length;
-							const parentsForQuery = parents.slice(i, endIndex);
-							const meshQuery = { _id: { $in: history.current }, shared_id: { $in: parentsForQuery }, type : "mesh"};
-							const meshProject = { shared_id: 1, _id: 0 };
-							queryPromise.push(dbCol.find(meshQuery, meshProject).toArray());
-						}
-
-						return Promise.all(queryPromise).then((res) => {
-							return res.reduce((acc, val) => acc.concat(val), []);
-						});
+/**
+ *
+ * @param {string} account
+ * @param {string} model
+ * @param {string} revId
+ */
+async function getIdToMeshesDict(account, model, revId) {
+	const treeFileName = `${revId}/idToMeshes.json`;
+	const {readStream: fileStream} = await FileRef.getJSONFileStream(account, model, treeFileName);
```

Comment:
Would it be better/faster to use getJSONFile instead of the streaming version? (No idea myself, genuine question)

Reply:
done